starlette/routing.py

```diff
@@ -33,11 +32,10 @@ class Match(Enum):
 def iscoroutinefunction_or_partial(obj: typing.Any) -> bool:
     """
     Correctly determines if an object is a coroutine function,
-    with a fix for partials on Python < 3.8.
+    by unwrapping functools.partial objects.
```

Comment:
This docstring is a bit too "how things work" rather than "what this does". Suggested:
```suggestion
    including those wrapped in functools.partial objects.
```

Reply:
Hey @JayH5, thanks for getting back to me on this!
I agree that docstring could definitely be more informative, should I link the #1107 issue as well?

Reply:
You could, but maybe in a code comment (it's an implementation detail) rather than in the docstring.